src/containers/announcements/ducks/thunks.ts

```diff
@@ -0,0 +1,19 @@
+import { AnnouncementProps, AnnouncementsThunkAction } from './types';
+import { announcements } from './actions';
+
+export const getAnnouncements = (limit?: number): AnnouncementsThunkAction<void> => {
+    return (dispatch, getState, { publicApiClient }) => {
+        dispatch(announcements.loading());
+        return publicApiClient
+            .getAnnouncements()
+            .then((response: AnnouncementProps[]) => {
+                return (limit && limit >= 0) ? response.slice(0, limit) : response;
```

Comment:
What does this logic do? Does this mean we're enforcing the limit client side? If we're loading all of the announcements, we should just store them all.